provider/openstack/provider.go

```diff
@@ -2115,10 +2115,55 @@ func (e *Environ) terminateInstances(ctx context.ProviderCallContext, ids []inst
 				break
 			}
 		}
+
+		// Attempt to destroy the ports that could have been created when using
+		// spaces.
+		if err := e.terminateInstanceNetworkPorts(id); err != nil {
```

Comment:
We need to do this first, because the ports will no longer be recognised by the instance ID.
They will be detached when the server is destroyed.